service/service.py

```diff
@@ -27,6 +37,57 @@ def index():
     """ Root URL response """
     return "Some useful information in json format about the recommendations service", status.HTTP_200_OK
 
+######################################################################
+# QUERY RELATED PRODUCTS BY ID
+######################################################################
+@app.route("/recommendations/<int:id>", methods=["GET"])
+def get_related_products(id):
+    """
+    Retrieve all related products by providing a product id
+    returns a list of 3 objects, showing a list of active ids and inactive ids for each realationship type
+    [
+        {
+            relationship-id: 1,
+            ids: [id1, id2, id3, ...],
+            inactive-ids: [id10, id20, id30, ...]
+        },
+        {
+            relationship-id: 2,
+            ids: [id4, id5, id6, ...],
+            inactive-ids: [id40, id50, id60, ...]
+        },
+        {
+            relationship-id: 3,
+            ids: [id7, id8, id9, ...],
+            inactive-ids: [id70, id80, id90, ...]
+        }
+    ]
+    """
+    app.logger.info("Request for related products with id: %s", id)
+    products = Recommendation.find(id) # need to replace find method with actual function name from model file
+    
+    if not products:
+        raise NotFound("Product with id '{}' was not found.".format(id))
+    
+    # assume model returns recors in format of: [{id: 1, rel_id: 2, typeid: 1, status: true}]
+    relationships = []
+    type_1_active, type_1_inactive = [], []
```

Comment:
Just for clarification, the relationship (`typeid`) are `0/1/2` right?

Reply:
No from the story, the typeid is 1/2/3, and that's what the code being implemented. If we agree to use 0/1/2 instead, that'd be an easy fix on the code though.

Reply:
Oh I just checked the story again and noticed that the `typeid` in the "assumption" part and in the "acceptance criteria" part are different - that's why I am confused 🤣 I guess we can all use 1/2/3.

Reply:
Yeah that shouldn't be a big deal since the data will be created through our service anyway. As long as we follow the same standard from now on we should be fine. I thought start form 1 would be a good idea, since we are using integer id, I'm not sure if id 0 could be interpreted as boolean false.

Reply:
Sure. I just changed the `typeid` range in my code to `1/2/3`.